cmd/auto-updater/main.go

```diff
@@ -167,37 +162,50 @@ func cmdRun(cmd *cobra.Command, args []string) (err error) {
 		return nil
 	}
 
-	err = loop.Run(ctx, func(ctx context.Context) (err error) {
+	loopInterval, err := time.ParseDuration(interval)
+	if err != nil {
+		return errs.New("unable to parse interval parameter: %v", err)
+	}
+
+	loopFunc := func(ctx context.Context) (err error) {
 		if err := update(ctx); err != nil {
 			// don't finish loop in case of error just wait for another execution
 			log.Println(err)
 		}
 		return nil
-	})
+	}
+
+	if loopInterval == 0 {
+		err = loopFunc(ctx)
```

Comment:
Why do we have this special case? Does it mean that if `loopInterval == 0` then the check or update will be executed only once and the program will exit?

Reply:
yes, with loopInterval == 0 program will execute only once. Its a case for automatic test.

Reply:
This sounds like an easter egg.

To me, loopInterval == 0 sound like a valid case for the Cycle to not wait between loop runs.

Can we check for a negative interval instead? If interval < 0 then run only once. And document it in the flag description.

Reply:
done

Reply:
Oh, shoot! I thought `0` is a valid value for `sync2.NewCycle`, but it is the `time.Ticker` that does not accept it. Then maybe check for `loopInterval <= 0` instead. Sorry about that!